File: core/src/main/java/com/canyon/commons/engines/CircleEngine.java
package com.canyon.commons.engines;

/**
		* 循环引擎
		*/
public interface CircleEngine {

	/**
	 * 开始
	 */
	void start();

	/**
	 * 停止
	 */
	void stop();

	/**
	 * 是否已经开始
	 * @return
	 */
	boolean getIsStart();

	/**
	 * 检查时间（单位：秒）
	 * @return
	 */
	int getDetectSpanInSecs();

	/**
	 * 设置检查时间（单位：秒）
	 * @param detectSpanInSecs
	 */
	void setDetectSpanInSecs(int detectSpanInSecs);

	/**
	 * 获取最后异常
	 * @return
	 */
	Exception getLastException();
}

File: commons/src/main/kotlin/com/canyon/commons/engines/CircleEngine.kt
package com.canyon.commons.engines

/**
 * 循环引擎
 */
interface CircleEngine {

    /**
     * 是否已经开始
     * @return
     */
    val isStart: Boolean

    /**
     * 检查时间（单位：秒）
     * @return
     */
    /**
     * 设置检查时间（单位：秒）
     * @param detectSpanInSecs
     */
    var detectSpanInSecs: Int

    /**
     * 获取最后异常
     * @return
     */
    val lastException: Exception?

    /**
     * 开始
     */
    fun start()

    /**
     * 停止
     */
    fun stop()
}
